### DISCUSSION TITLE
Issue with Write Image (ITK) Filter

### DISCUSSION THREAD

**Discussion by maliesha95** [BlueQuartzSoftware/DREAM3DNX-Issues#12] [Q&A] [answered]
Hi, 

I'm new to DREAM3D and trying to run the prebuilt pipeline called "TxCopper_Exposed". But the application crashes at the Write Image (ITK) step (previous filters in the pipeline run without any issues) with the following error in the terminal (Blue box at the end). I'm also including the log from when I started the application, as I'm not sure if the errors during startup are related to the Write Image (ITK) Filter issue.

![Screenshot_20240415_130110](https://github.com/BlueQuartzSoftware/DREAM3DNX-Issues/assets/135638612/da4a2821-c9fb-407f-b19e-53d70c1daa7d)

The below screenshot shows the Filter Parameters of the Write Image (ITK).
![Screenshot_20240415_130202](https://github.com/BlueQuartzSoftware/DREAM3DNX-Issues/assets/135638612/b54cedb0-64b3-4719-9d4a-8f55ac876a53)

I would really appreciate your help with resolving this issue and saving the maps without any issues.

Thanks.

**Comment by imikejackson** [comment]
Hello @maliesha95 I am a little bit perplexed by this error. It looks like /tmp is symlinked to another physical device on your system? Maybe? I have not seen this kind of error before. We will take a look into the error and see if we can find something.

@nyoungbq

**Comment by nyoungbq** [comment]
@maliesha95 So I have identified the issue. It seems that the system you are working on has a home directory mounted as its own filesystem and the file that is being written is mounted on the root filesystem's temporary directory. The issue occurs when the temp file is trying to be moved to its final output location. This is not a fault on your end.

The bug is in the ITKImageWriter itself, I will get a fix in immediately, so it will no longer crash.

### Temporary Fixes
The temporary fix for now is to grab the file out of your temp directory. It was written out there and it should exist. However, if you want to visualize the data immediately and you don't need that exact file format, I suggest trying another one of our writers in the application as they have been designed more robustly and should accommodate your system configuration, based on the code I looked at.

**Comment by nyoungbq** [comment]
This PR contains the fix: https://github.com/BlueQuartzSoftware/simplnx/pull/918

**Comment by nyoungbq** [comment, marked as answer]
The commit containing the fix was included in the RC-11 release, so I am going to close the discussion. Thanks for taking the time to report it! Be sure to reach out or reopen the discussion if the issue persists.
